File: Medium/Python/ArrayAndStrings/IncreasingTriplet.py
class Solution(object):
    def increasingTriplet(self, nums):
        """
            :type nums: List[int]
            :rtype: bool
            """
        L = [float("inf"), float("inf"), float("inf")]
        if len(nums) >= 3:
            for i in range(len(nums)):
                if nums[i] < L[0]:
                    L[0] = nums[i]
                elif nums[i] > L[0]:
                    if nums[i] < L[1]:
                        L[1] = nums[i]
                    elif nums[i] > L[1]:
                        return True
        return False

File: Medium/Java/ArrayAndStrings/IncreasingTriplet.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        if (nums.length >= 3) {
            int[] trip = {Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE};
            for (int num: nums) {
                if (num > trip[0]) {
                    if (num > trip[1])
                        return true;
                    trip[1] = num;
                }
                else
                    trip[0] = num;
            }
        }
        return false;
    }
}
